src/metabase/shared/automagic_dashboards/constants.cljc

```diff
@@ -0,0 +1,34 @@
+(ns metabase.shared.automagic-dashboards.constants)
+
+(def ^:export GRID_WIDTH
+  "Default width of a dashboard"
+  24)
+
+(def ^:private card-size-defaults*
+  {:table       {:min {:width 4 :height 3} :default {:width 12 :height 9}}
+   :gauge       {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :bar         {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :pie         {:min {:width 4 :height 3} :default {:width 12 :height 8}}
+   :scatter     {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :waterfall   {:min {:width 4 :height 3} :default {:width 14 :height 6}}
+   :combo       {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :stacked     {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :scalar      {:min {:width 2 :height 2} :default {:width 6 :height 3}}
+   :line        {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :link        {:min {:width 1 :height 1} :default {:width 8 :height 1}}
+   :action      {:min {:width 1 :height 1} :default {:width 4 :height 1}}
+   :area        {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :pivot       {:min {:width 4 :height 3} :default {:width 12 :height 9}}
+   :funnel      {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :progress    {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :smartscalar {:min {:width 2 :height 2} :default {:width 6 :height 3}}
+   :map         {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :object      {:min {:width 4 :height 3} :default {:width 12 :height 9}}
+   :row         {:min {:width 4 :height 3} :default {:width 12 :height 6}}
+   :heading     {:min {:width 1 :height 1} :default {:width GRID_WIDTH}}
+   :text        {:min {:width 1 :height 1} :default {:width 12 :height 3}}})
+
+(def ^:export card-size-defaults
+  "Default card sizes per visualization type"
+  #?(:clj  card-size-defaults*
+     :cljs (clj->js card-size-defaults*)))
```

Comment:
I'd do this same-data-different-format magic the other way around. Rename `card-size-defaults*` to remove the star - that's the symbol for use from **all** Clojure code. Then add
```clojure
#?(:cljs (def ^:export card-size-defaults-json (clj->js card-size-defaults)))
```
and import that from TS.

I'd certainly be surprised if I were working in some CLJC file, `:require`d some CLJC namespace, and received JS data! We use the approach I'm suggesting in MLv2: most of the files are all Clojure all the time and have very few `#?(:cljs ...)` conditionals, then `metabase.lib.js` wraps many of those functions with extra JS conversion for inputs and outputs.

Reply:
That make sense.